pandas/core/generic.py

```diff
@@ -1940,6 +1940,78 @@ def tail(self, n=5):
             return self
         return self.iloc[-n:]
 
+  
+    def sample(self, n=None, frac=None, replace=False, weights=None, seed=None):
+        """
+        Returns a random sample of rows from object. 
+        
+        Parameters
+        ----------
+        n : int, optional
+            Number of rows to return. Cannot be used with frac.
+            Default = 5 if frac = None. 
+        frac : float, optional
+            Share of rows to return. Cannot be used with n. 
+        replace : boolean, optional
+            Sample with or without replacement. Default = False. 
+        weights : str or ndarray-like, optional
+            Default 'None' results in equal probability weighting. 
+            If called on a DataFrame or Panel, will also accept the name of a 
+            column as a string. Must be same length as index. 
+            Weights must be normalized. 
+        seed : int, optional
+            Seed for the random number generator.
+            
+        Returns
+        -------
+        Same type as caller. 
+        -----------
+        """
+    
+        # Setup vars
+        rs = np.random.RandomState(seed)
+    
+        # Check weights for compliance
+        if weights is not None:
+                                      
+            # Strings acceptable if not a series
+            if isinstance(weights, string_types): 
+
+                if self.ndim > 1 :
+                    try:
+                        weights = self[weights]
+                    except KeyError:
+                        raise KeyError("String passed to weights not a valid column name")
+
+                else:
+                    raise ValueError("Strings cannot be passed as weights when sampling from a Series.")
+
+            #normalize format of weights to ndarray. 
+            weights = np.asarray(weights)
+                
+            # Check length (numpy does this, but has confusing errors with different argument labels.)
+            if len(weights) != len(self):
+                raise ValueError("Weights and data to be sampled must be of same length")
+
+    
+        # Check whether frac or N
+        if n is None and frac is None:
+            n = 5
+        elif n is not None and frac is None and n % 1 != 0:
+            raise ValueError("Only integers accepted as `n` values")
+        elif n is None and frac is not None:
+            n = int(round(frac * len(self))) 
+        elif n is not None and frac is not None:
+            raise ValueError('Please enter a value for `frac` OR `n`, not both')
+    
+        # Check for negative sizes
+        if n < 0:
+            raise ValueError("A negative number of rows requested. Please provide positive value.")
+    
+        locs = rs.choice(len(self), size = n, replace = replace, p = weights)
```

Comment:
Please double check PEP 8 -- you should not be using spaces around `=` in function arguments :). This should be `rs.choice(len(self), size=n, replace=replace, p=weights)`, and the same elsewhere.